### DISCUSSION TITLE
Issue (maybe) with $money and Intl

### DISCUSSION THREAD

**Discussion by cfjedimaster** [alpinejs/alpine#3746] [1. Help]
I wasn't sure if I should file this under bugs as it may be by design. I'm playing with the $money mask and it works great, but I ran into an issue trying to make it more dynamic. $money lets you specify an optional second argument for the decimal character and a third argument for thousands separator. This works fine when hard coded. However, I wanted to  make use of the Intl API to get the user's locale and formatting rules, and then apply that dynamically. So for example:

```html
<input id="test4" x-mask:dynamic="$money($input,this.decimal,this.group)">
```

And in my code, I've got a function to get the user's locale, and a function that gets the decimal and group values from the Intl spec for the current locale. 

```js
init() {
	let separators = getSeparators(this.locale);
//	this.decimal = separators.decimal;
//	this.group = separators.group;
	this.group = 'y';
	this.decimal = 'x';
}
```

I hard coded x and y to help me test. When run, it appears as if the values are not used, and defaults get applied instead (I get "," and ".") Is it possible for the second and third arguments to be dynamic?

I also tried changing $money to a function that x-mask would call, but in that function, you don't have access to $money.

I also tried hiding the mask until I had gotten the two values, that didn't seem to help. 

CodePen here: https://codepen.io/cfjedimaster/pen/OJrRWVe?editors=1011

**Comment by trych** [comment]
This line

```html
x-mask:dynamic="$money($input,this.decimal,this.group)"
```

needs to be

```html
x-mask:dynamic="$money($input,decimal,group)"
```

decimal and group are just part of your regular data.

**Reply by cfjedimaster** [reply]
Ugh - and I've been doing Alpine for quite some time and just brain farted on that. Thank you!
